aws/provider_test.go

```diff
@@ -750,3 +749,21 @@ data "aws_arn" "test" {
 }
 `, region)
 }
+
+func testAccAssumeRoleARNPreCheck(t *testing.T) {
+	v := os.Getenv("TF_ACC_ASSUME_ROLE_ARN")
+	if v == "" {
+		t.Skip("skipping tests; TF_ACC_ASSUME_ROLE_ARN must be set")
+	}
+}
+
+func testAccProviderConfigAssumeRolePolicy(policy string) string {
```

Comment:
This is a really neat setup. Do you have time to document this awesome new way of "acceptance testing resources with restricted IAM permissions" in the contributing guide? I'm sure there are other resources that could benefit from something like this. ❤️